### PR TITLE
CMF-8:Menu Item CRUD

### PR DESCRIPTION
**Pull request by CrooZER**
- added menu item crud

### CODE REVIEW

File: Entity/BaseMenuItem.php [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+     * @var \DateTime
+     */
+    protected $updatedAt;
+
+    /**
+     * @var BaseMenu
+     */
+    protected $menu;
+
+
```
Comment (aivus) [inline Entity/BaseMenuItem.php:79]: Please, remove excessive line

File: DependencyInjection/GrossumMenuExtension.php [hunk 2]
```diff
@@ -38,5 +39,39 @@ public function load(array $configs, ContainerBuilder $container)
...
+    }
+
+    /**
+     * @param array $config
+     */
+    public function registerDoctrineMapping(array $config)
+    {
+        $collector = DoctrineCollector::getInstance();
+        $collector->addAssociation($config['class']['menu_item'], 'mapManyToOne', array(
+            'fieldName'     => 'contact',
```
Comment (aivus) [inline DependencyInjection/GrossumMenuExtension.php]: We don't have `contact` field in `MenuItem` class

File: DependencyInjection/GrossumMenuExtension.php [hunk 3]
```diff
@@ -38,5 +39,39 @@ public function load(array $configs, ContainerBuilder $container)
...
+            'targetEntity'  => $config['class']['menu'],
+            'cascade'       => array(
+                'persist',
+            ),
+            'mappedBy'      => null,
+            'inversedBy'    => 'menu_items',
```
Comment (aivus) [inline DependencyInjection/GrossumMenuExtension.php]: It should be property name in class `Menu`

File: DependencyInjection/GrossumMenuExtension.php (hunk 2 above)
Comment (aivus) [inline DependencyInjection/GrossumMenuExtension.php]: I think it should be `menu`

File: Entity/EntityManager/BaseMenuManager.php [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+     */
+    private $repository;
+
+    /** @var  ObjectManager */
+    private $objectManager;
+
+    public function __construct(ObjectManager $objectManager, $menuClass)
+    {
+        $this->objectManager = $objectManager;
+        $this->repository    = $objectManager->getRepository('GrossumMenuBundle:BaseMenu');
```
Comment (aivus) [inline Entity/EntityManager/BaseMenuManager.php]: Please, remove extra whitespaces

File: Entity/EntityManager/BaseMenuManager.php [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+     */
+    private $repository;
+
+    /** @var  ObjectManager */
+    private $objectManager;
+
+    public function __construct(ObjectManager $objectManager, $menuClass)
+    {
+        $this->objectManager = $objectManager;
+        $this->repository = $objectManager->getRepository('GrossumMenuBundle:BaseMenu');
```
Comment (CheAlex) [inline Entity/EntityManager/BaseMenuManager.php]: Please remove this line - you are already using lazy load below

File: Entity/BaseMenuItem.php [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+     * @var BaseMenu
+     */
+    protected $menu;
+
+    /**
+     * Set name
+     *
+     * @param string $name
+     *
+     * @return BaseMenuItem
```
Comment (CheAlex) [inline Entity/BaseMenuItem.php]: Please replace all annotations `@return BaseMenuItem` to `@return $this`. This allow to use autocomplete in class-children

File: Entity/BaseMenuItem.php [hunk 7]
```diff
@@ -0,0 +1,125 @@
+<?php
+
+namespace Grossum\MenuBundle\Entity;
+
+/**
+ * BaseMenuItem
+ */
```
Comment (CheAlex) [inline Entity/BaseMenuItem.php]: Please remove class-annotation if there is no important information

File: Entity/EntityManager/BaseMenuManager.php [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+     */
+    public function getRepository()
+    {
+        if ($this->repository === null) {
+            $this->repository = $this->objectManager->getRepository($this->menuClass);
+        }
+
+        return $this->repository;
+    }
+}
```
Comment (WhiteEagle88) [inline Entity/EntityManager/BaseMenuManager.php:51]: If you create manager, why you don't announced in services?

Comment (aivus) [inline Entity/EntityManager/BaseMenuManager.php:51]: What do you mean?

**Pull request merged**